backend/src/core/services/stats.service.ts

```diff
@@ -1,19 +1,102 @@
-import { fn, cast, Transaction, Op } from 'sequelize'
+import { fn, cast, Transaction, Op, QueryTypes } from 'sequelize'
 import { Statistic, JobQueue, Campaign } from '@core/models'
 import {
   CampaignStats,
   CampaignStatsCount,
   CampaignInvalidRecipient,
 } from '@core/interfaces'
-import { MessageStatus } from '@core/constants'
+import { MessageStatus, ChannelType } from '@core/constants'
+import logger from '@core/logger'
+
+import bluebird from 'bluebird'
+
+const getUnfinalisedCampaigns = async (): Promise<Array<number>> => {
+  const result = await Statistic.sequelize?.query(`
+      SELECT id FROM campaigns WHERE id NOT IN (
+        SELECT DISTINCT(campaign_id) FROM email_messages
+        WHERE
+          updated_at::TIMESTAMP > (clock_timestamp() - '36 hours'::INTERVAL)
+          AND status IS NOT NULL
+      ) AND halted <> TRUE;`)
+  return result?.[0].map((r: any) => r.id as number) || []
+}
+
+/**
+ * Calls update_stats SQL function for respective channels.
+ * fails silently
+ * @param campaignId
+ */
+const updateStats = async (campaignId: number): Promise<void> => {
+  const campaign = await Campaign.findOne({
+    where: { id: campaignId },
+  })
+  if (!campaign) return
+
+  logger.info(`updateStats invoked for campaign ${campaignId}`)
+
+  switch (campaign.type) {
+    case ChannelType.Email: {
+      await Statistic.sequelize?.query(
+        'SELECT update_stats_email(:campaign_id);',
+        {
+          replacements: { campaign_id: campaignId },
+          type: QueryTypes.SELECT,
+        }
+      )
+      break
+    }
+    case ChannelType.SMS: {
+      await Statistic.sequelize?.query(
+        'SELECT update_stats_sms(:campaign_id);',
+        {
+          replacements: { campaign_id: campaignId },
+          type: QueryTypes.SELECT,
+        }
+      )
+      break
+    }
+    case ChannelType.Telegram: {
+      await Statistic.sequelize?.query(
+        'SELECT update_stats_telegram(:campaign_id);',
+        {
+          replacements: { campaign_id: campaignId },
+          type: QueryTypes.SELECT,
+        }
+      )
+      break
+    }
+    default: {
+      logger.error(`Invalid channel type found for campaign ${campaignId}`)
+    }
+  }
+}
+
+const consolidateStats = async (): Promise<void> => {
```

Comment:
Is this unused? why?

Reply:
Oh unused for now. This was the query I wanted to use to fetch campaigns that stats that have not been updated for the last 36 hours. 

Actually let me go ahead and remove this, since we now want a job to be scheduled for X hours after the campaign is sent.